src/main/java/net/fabricmc/loader/game/MinecraftGameProvider.java

```diff
@@ -221,10 +222,10 @@ public void launch(ClassLoader loader) {
 		}
 
 		try {
-			Class<?> c = loader.loadClass(targetClass);
-			Method m = c.getMethod("main", String[].class);
-			m.invoke(null, (Object) arguments.toArray());
-		} catch (Exception e) {
+			MethodHandles.lookup()
+					.findStatic(loader.loadClass(targetClass), "main", MethodType.methodType(Void.TYPE, String[].class))
+					.invokeExact((String[]) arguments.toArray()); // ignore ide warning here
```

Comment:
Our static analysis engine detects that your code is problematic. According to the [Correctness](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [BC: Impossible downcast of toArray() result (BC_IMPOSSIBLE_DOWNCAST_OF_TOARRAY)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bc-impossible-downcast-of-toarray-result-bc-impossible-downcast-of-toarray).

This code is casting the result of calling `toArray()` on a collection to a type more specific than `Object[]`, as in:
```java
String[] getAsArray(Collection<String> c) {
    return (String[]) c.toArray();
}
```
This will usually fail by throwing a **ClassCastException**. The `toArray()` of almost all collections return an `Object[]`. They can't really do anything else, since the Collection object has no reference to the declared generic type of the collection.

The correct way to do get an array of a specific type from a collection is to use `c.toArray(new String[]);` or `c.toArray(new String[c.size()]);` (the latter is slightly more efficient).

There is one common/known exception to this. The `toArray()` method of lists returned by `Arrays.asList(...)` will return a covariantly typed array. For example, `Arrays.asArray(new String[] { "a" }).toArray()` will return a `String []`.

Reply:
Oh my god, when can this spam stop